src/engine/serverbrowser.cpp

```diff
@@ -273,32 +253,35 @@ void pingservers()
         enet_socket_set_option(pingsock, ENET_SOCKOPT_NONBLOCK, 1);
         enet_socket_set_option(pingsock, ENET_SOCKOPT_BROADCAST, 1);
     }
-    ENetBuffer buf;
+    int tmp_millis = totalmillis;
     uchar ping[MAXTRANS];
+    // ENet is crappy enough to invert it's members depending on
+    // target OS.
+    // This does also imply that this "efficient" library wastes 25%
+    // bytes in alignment on windows for this structure.
+#ifdef WIN32
+    ENetBuffer buf = { 0, ping };
+#else
+    ENetBuffer buf = { ping, 0 };
+#endif
```

Comment:
Why not do it like this, like we do in other places in the code?

```
    ENetBuffer buf;
    buf.data = ping;
    buf.dataLength = 0;
```

Then we don't need to care about the order.

Reply:
```cpp
    ENetBuffer buf{};
```

Otherwise, the struct isn't initialized properly. But I'd just

```cpp
    ENetBuffer buf{0, ping};
```

Reply:
There are two problems with `ENetBuffer buf{0, ping};`:

1. We need to write different code for different operating systems as seen in this code.
2. We can't see that the right members get the right values; we have to rely on knowing the order. I like the designated initializers in C99 but if we don't have them I think the best we can do, if we want to make sure everything is initialized, is:

```
    ENetBuffer buf{};
    buf.data = ping;
    buf.dataLength = 0;
```

Reply:
Well, I'm not a big fan of relying on the order either, and even if it's rather unlike that things might go wrong with regards to that, it's also important to keep things consistent in the code. Some day, we might make ENet a more usable C++ library, but until this day has come, let's just keep things consistent.

We should really turn on this warning in the compiler that complains about uninitialized structs...